projects/solar/src/relay_fsm.c

```diff
@@ -2,17 +2,64 @@
 
 #include <stdbool.h>
 
+#include "can_transmit.h"
+#include "can_unpack.h"
+#include "data_store.h"
 #include "event_queue.h"
 #include "fault_handler.h"
 #include "fsm.h"
+#include "gpio.h"
 #include "log.h"
+#include "solar_boards.h"
 #include "solar_config.h"
 #include "solar_events.h"
 #include "status.h"
 
+static SolarMpptCount solar_mppt_count;
+
+static void prv_assert_relay(SoftTimerId timer_id, void *context) {
+  RelayFsmStorage *storage = (RelayFsmStorage *)context;
```

Comment:
Nit: since `context` is a void pointer, you don't actually need the cast here.

Reply:
do I have to do like `(RelayFsmStorage *)context)->is_error_called` since just doing `context->is_error_called` throws me 
expression must have pointer-to-struct-or-union type but it has type "void *" error

Reply:
No what I meant is you can just do `RelayFsmStorage *storage = context` without the cast.

Reply:
done